src/blueprint/README.md

```diff
@@ -0,0 +1,153 @@
+```
+az extension add --name blueprint
+```
+
+Blueprint can be scoped in a subscription or management group, which is a group of subscriptions. You can use management group by specify `--management-group`, or use subscription by specify `--subscription.` If both parameters are omitted, the command will use your default subscription. You can set it by running:
+```
+az account set --subscription subscription_id
+```
+
+### Included Features
+#### Blueprint Definition Management:
+*Examples:*
+
+##### Import Blueprint Definition and Artifacts Settings
+
+```
+az blueprint import \
+--name blueprintName \
+--input-path "/path/to/blueprint/directory"
+
+```
+
+In the input directory, there should be a file named "blueprint.json" that has blueprint definition and parameters for artifacts. There should be a subdirectory named "artifacts" and it should contain files for artifact definition. Examples can be found [here]().
+
+The import command will overwrite defnitions for blueprint and artifacts if a blueprint with the same name already exists.
+
+##### Create a Blueprint with Parameters
+
+```
+az blueprint create \
+    --name blueprintName \
+    --description "An example blueprint." \
+    --target-scope "subscription" \
+    --parameters @/path/to/blueprint_params.json
+```
+An example blueprint_params.json may look like this:
+```json
+{
+    "MyRoleName_RoleAssignmentName": {
+        "type": "array",
+        "displayName": "[User group or application name] ([User group or application name] : Reader)",
+        "strongType": "PrincipalId"
+    },
+    "MyPolicyName_Members": {
+      "type": "string",
+      "displayName": "Members",
+      "allowedValues": []
+    }
+}
+```
+The paramters are for artifacts which will be added in below commands.
+
+##### Add a Resource Group in the Blueprint
+```
+az blueprint resource-group create \
+    --blueprint-name blueprintName \
+    --artifact-name myRgArt
+```
+
+##### Add a Role Assignment Artifact
+```
+az blueprint artifact role create \
+    --blueprint-name blueprintName \
+    --artifact-name my-role-art \
+    --display-name "My Role Name" \
+    --resource-group-art myRgArt \
+    --role-definition-id "/providers/Microsoft.Authorization/roleDefinitions/00000000-0000-0000-0000-000000000000" \
+    --principal-ids "[parameters('MyRoleName_RoleAssignmentName')]"
+```
+
+##### Add a Policy Assignment Artifact
+```
+az blueprint artifact policy create \
+    --blueprint-name blueprintName \
+    --artifact-name my-policy-art \
+    --display-name "My Policy Name" \
+    --policy-definition-id "/providers/Microsoft.Authorization/policySetDefinitions/00000000-0000-0000-0000-000000000000" \
+    --parameters @/path/to/policy_params.json
+```
+An example policy_params.json may look like this:
+```json
+{
+    "Members":{
+        "value":"[parameters('MyPolicyName_Members')]"
+    }
+}
+```
+
+
+##### Publish a Blueprint
+```
+az blueprint published create \
```

Comment:
just to confirm, published will be renamed to publish

Reply:
will just use `az blueprint publish` for this command.